### PR TITLE
LPS-138756 Specs: when there is no client registration feature,

### PR DESCRIPTION
**Pull request by arthurchan35**
the algorithm for signature should be the default of RS256

a future task will be to add the feature of client registration

### CODE REVIEW

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by stian-sigvartsen** [comment]
Hey @arthurchan35 . I've pushed a commit for the whitelisting we discussed. Can you have a look?
If it is fine then we will also need to update the language keys / documentation to reflect the new logic.

File: modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectMetadataFactoryImpl.java [hunk 1]
```diff
@@ -189,6 +209,22 @@ protected synchronized void refresh(long time)
...
+			if (_allowedIdTokenSigningAlgValues != null) {
+				List<JWSAlgorithm> idTokenJWSAlgs =
+					_oidcProviderMetadata.getIDTokenJWSAlgs();
+
+				Stream<JWSAlgorithm> stream = idTokenJWSAlgs.stream();
+
+				_oidcProviderMetadata.setIDTokenJWSAlgs(
+					stream.filter(
+						jwsAlgorithm ->
+							_allowedIdTokenSigningAlgValues.contains(
```
Comment (arthurchan35) [inline modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectMetadataFactoryImpl.java:221]: @stian-sigvartsen 
The filtering here does not make sense. The refresh logic here is for re-discover provider metadata info, a filtering here means to filter returned supported algorithms against configured supported algorithms.
A simple test will be if you configure both RS256, HS256 in configuration page, then you will fail algorithm check again.

The filtering place I was refering to is in requestUtil's validate method, where you currently choose header's algorithm to validate.
What we need to do is beforing choosing header's algorithm to validate, we verify header's algorithm against the supported algorithms.

**Comment by arthurchan35** [comment]
> Hey @arthurchan35 . I've pushed a commit for the whitelisting we discussed. Can you have a look?
> If it is fine then we will also need to update the language keys / documentation to reflect the new logic.

Hey Stian, I don't think there is any lang keys needing update, the current description for:

ID Token Signning Algorithm entry in configuration page is:

> Set the supported ID token signing algorithms for the OpenID Connect Provider. If the discovery endpoint is not specified, these algorithms are used. In this case, the algorithm RS256 must be included and must be the first algorithm in the list. If the discovery endpoint is specified, no algorithms should be specified here since the supported signing algorithms for the identity token are obtained from the discovery endpoint.

Whose purpose is very clear. Like I mentioned, when provider does not have discovery endpoint/client does not want to use discovery endpoint, client can manually copy provider metadata info here.

**Comment by arthurchan35** [comment]
https://github.com/liferay-appsec/liferay-portal/pull/496

**Pull request closed without merging**